plugins/javaembed/javaembed.cpp

```diff
@@ -1597,11 +1601,11 @@ class JavaThreadContext
         return contextClassLoaderObj;
     }
 
-    void setThreadClassLoader(const char *classPath)
+    void setThreadClassLoader(const char *classPath, size32_t bytecodeLen, const byte *bytecode)
     {
-        if (classPath && *classPath)
+        if (bytecodeLen || (classPath && *classPath))
         {
-            if (prevClassPath && strcmp(classPath, prevClassPath) == 0)
+            if (!bytecodeLen && prevClassPath && classPath && strcmp(classPath, prevClassPath) == 0)  // MORE - caching of inline classes is important too...
```

Comment:
Is there a jira to improve the caching?  Is it correct that precompiled is not currently cached?

Reply:
This will be moot if I manage to implement the thread-local stuff. If I can't I will have to revisit.